src/txadmission.cpp

```diff
@@ -896,7 +925,23 @@ bool ParallelAcceptToMemoryPool(Snapshot &ss,
             if (!pool._CalculateMemPoolAncestors(entry, setAncestors, nLimitAncestors, nLimitAncestorSize,
                     nLimitDescendants, nLimitDescendantSize, errString))
             {
-                return state.DoS(0, false, REJECT_NONSTANDARD, "too-long-mempool-chain", false, errString);
+                // If the chain is not sync'd entirely then we'll defer this tx until the new block is processed.
+                if (!IsChainSyncd() && IsChainNearlySyncd())
+                {
+                    CTxInputData txd;
+                    txd.tx = tx;
+                    txd.nodeId = -1;
+                    txd.nodeName = "deferring";
+                    txd.whitelisted = false;
+                    LOG(MEMPOOL, "Deferring - too many ancestors tx: %s\n", tx->GetHash().ToString());
+                    fWaitQ = true;
+
+                    LOCK(csTxInQ);
+                    txWaitQ.push(txd);
```

Comment:
We should limit the size of this Wait Q to stop a memory exhaustion attack.

Reply:
I guess we have to pick a number...  something tied to EB would be best I think...maybe 10 txn per MB ...so currently would be 32 * 10.

Reply:
done